### PR TITLE
Changes per meeting with Ops team - Closes #624

### LINKED ISSUE #624: VRMS v0.3 Hosting Requests from Ops team

### PR DESCRIPTION
**Pull request by jasonwong26**
Met with Darren and the Ops team, and they asked for a few changes so they can move forward with hosting v0.3:

- Provide them with all necessary ENVIRONMENT variables (DONE)
- Convert settings for `MailHog` to ENVIRONMENT variables
- Set up new Production specific Docker files (`backend/Dockerfile.prod` and `client.Dockerfile.prod`)
- Create a set of health check pages (`/healthcheck` and `api/healthcheck`)

## New Environment Variables for MailHog
These changes require the following be added to the backend `.env` file:

```
MAILHOG_PORT=1025
MAILHOG_USER=user
MAILHOG_PASSWORD=password
```

We'll also want to update the Environment variables document once these changes are approved.

## New Docker Files for Dev

- `backend/Dockerfile.dev`
-  `client.Dockerfile.dev`

Darren recommended that we simplify the Dev environment with these (to replace the original `.api` and `.client` files), however I got `nginx` errors when I tested them.  
I kept them in this PR for reference, but we'll need additional changes to convert over.

### CODE REVIEW

**Comment by jasonwong26** [comment]
@alex-anakin, @mrao2 

I went ahead and added the environment variables to the Repository Secrets and PR Validation Workflow, but the ` test-backend-integration` still fails, reporting missing variables.

Do either of you know what I'm missing?  I need to stop for the night.

File: backend/Dockerfile.dev [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM node:14.11.0 AS api-development
+RUN mkdir /srv/backend
+WORKDIR /srv/backend
+RUN mkdir -p node_modules
+COPY package.json yarn.lock ./
+RUN yarn install --pure-lockfile
+COPY . .
```
Comment (alex-anakin) [inline backend/Dockerfile.dev]: please, add an empty line at the end of the file
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: backend/controllers/healthCheck.controller.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const HealthCheckController = {};
+
+HealthCheckController.isAlive = (_, res) => {
+  res.status(200).send("I'm Alive!");
+}
+
+module.exports = HealthCheckController;
```
Comment (alex-anakin) [inline backend/controllers/healthCheck.controller.js]: please, add an empty line at the end of the file
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: client/Dockerfile.dev [hunk 3]
```diff
@@ -0,0 +1,7 @@
+FROM node:14.11.0 AS client-development
+RUN mkdir /srv/client && chown node:node /srv/client
+WORKDIR /srv/client
+USER node
+RUN mkdir -p node_modules
+COPY --chown=node:node package.json package.json ./
+RUN npm install --silent
```
Comment (alex-anakin) [inline client/Dockerfile.dev]: please, add an empty line at the end of the file
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

**Review by alex-anakin — CHANGES_REQUESTED** [review]
Everything is fine except for a couple of missed new lines at the end of the file.

**Comment by alex-anakin** [comment]
@jasonwong26 Could you try please to close this PR and create a new one? For me, it seems like the PR "remembers" the state of actions that were run for it from the beginning. And it does not refresh the state after updating the action.

**Comment by jasonwong26** [comment]
Per @alex-anakin request, will close and reopen this PR.

**Pull request closed without merging**